src/tests/emote.cairo

```diff
@@ -0,0 +1,59 @@
+// Core imports
+
+use core::debug::PrintTrait;
+
+// Starknet imports
+
+use starknet::testing::set_contract_address;
+
+// Dojo imports
+
+use dojo::world::{IWorldDispatcher, IWorldDispatcherTrait};
+
+// Internal imports
+
+use zconqueror::config;
+use zconqueror::store::{Store, StoreTrait};
+use zconqueror::models::game::{Game, GameTrait};
+use zconqueror::models::player::Player;
+use zconqueror::models::tile::Tile;
+use zconqueror::systems::host::IHostDispatcherTrait;
+use zconqueror::systems::play::IPlayDispatcherTrait;
+use zconqueror::tests::setup::{setup, setup::{Systems, Context, HOST, PLAYER}};
+
+// Constants
+
+const HOST_NAME: felt252 = 'HOST';
+const PLAYER_NAME: felt252 = 'PLAYER';
+const PRICE: u256 = 1_000_000_000_000_000_000;
+const PENALTY: u64 = 60;
+const PLAYER_COUNT: u8 = 2;
+const PLAYER_INDEX: u32 = 0;
+
+
+#[test]
+#[available_gas(1_000_000_000)]
+#[should_panic(expected: ('Emote: invalid player', 'ENTRYPOINT_FAILED',))]
+fn test_emote_revert_invalid_player() {
```

Comment:
Nice test but please add a valid one in addition to this one.

Reply:
I'm done with testing ...